catt/api.py

```diff
@@ -11,56 +11,60 @@
 def discover() -> list:
     """Perform discovery of devices present on local network, and return result."""
 
-    return [CattDevice(ipaddr=d.host) for d in get_chromecasts()]
+    return [CattDevice(ip_addr=d.host) for d in get_chromecasts()]
 
 
 class CattAPIError(Exception):
     pass
 
 
 class CattDevice:
-    def __init__(self, name: str = "", ipaddr: str = "", lazy: bool = False) -> None:
+    def __init__(self, name: str = "", ip_addr: str = "", lazy: bool = False) -> None:
         """
         Class to easily interface with a ChromeCast.
 
         :param name: Name of ChromeCast device to interface with.
                      Either name of ip-address must be supplied.
-        :param ipaddr: Ip-address of device to interface with.
+        :param ip_addr: Ip-address of device to interface with.
                        Either name of ip-address must be supplied.
         :param lazy: Postpone first connection attempt to device
                      until first playback action is attempted.
         """
 
-        if not name and not ipaddr:
+        if not name and not ip_addr:
             raise CattAPIError("neither name nor ip were supplied")
+
         self.name = name
-        self.ipaddr = ipaddr
+        self.ip_addr = ip_addr
+        self.uuid = None
 
         self._cast = None
         self._cast_controller = None
         if not lazy:
             self._create_cast()
 
     def __repr__(self) -> str:
-        return "<CattDevice: %s>" % (self.name or self.ipaddr)
+        return "<CattDevice: %s>" % (self.name or self.ip_addr)
 
     def _create_cast(self) -> None:
-        self._cast = get_chromecast_with_ip(self.ipaddr) if self.ipaddr else get_chromecast(self.name)
+        self._cast = get_chromecast_with_ip(self.ip_addr) if self.ip_addr else get_chromecast(self.name)
         if not self._cast:
             raise CattAPIError("device could not be found")
         self._cast.wait()
+
         self.name = self._cast.name
-        self.ipaddr = self._cast.host
+        self.ip_addr = self._cast.host
+        self.uuid = self._cast.uuid
 
-    def _create_ctrl(self) -> None:
+    def _create_controller(self) -> None:
         self._cast_controller = get_controller(self._cast, get_app_info("default"))
 
     @property
-    def ctrl(self):
+    def controller(self):
```

Comment:
I would prefer ```control``` over ```controller```, as it's more distinct from the slightly confusing controller/castcontroller "paradigm".

Reply:
`control` is a verb though, implying a function. Or does it return "a control"?

Reply:
> Or does it return "a control"?

That is what I meant, I see your point though, about it being a verb. I just dont' like ```controller```.

Reply:
Yeah... We can find a completely different name, it doesn't have to be `controller`.